contracts/bridge/Bridge.sol

```diff
@@ -84,73 +84,128 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
     );
 
     // start allows the value transfer request.
-    function start() external onlyOwner {
-        isRunning = true;
-    }
-
-    // stop prevent the value transfer request.
-    function stop() external onlyOwner {
-        isRunning = false;
+    function start(bool _status)
+        external
+        onlyOwner
+    {
+        isRunning = _status;
     }
 
     // stop prevent the value transfer request.
-    function setCounterPartBridge(address _bridge) external onlyOwner {
+    function setCounterPartBridge(address _bridge)
+        external
+        onlyOwner
+    {
         counterpartBridge = _bridge;
     }
 
     // registerToken can update the allowed token with the counterpart token.
-    function registerToken(address _token, address _cToken) external onlyOwner {
+    function registerToken(address _token, address _cToken)
+        external
+        onlyOwner
+    {
         allowedTokens[_token] = _cToken;
     }
 
     // deregisterToken can remove the token in allowedToken list.
-    function deregisterToken(address _token) external onlyOwner {
+    function deregisterToken(address _token)
+        external
+        onlyOwner
+    {
         delete allowedTokens[_token];
     }
 
-    // handleERC20Transfer sends the token by the request.
-    function handleERC20Transfer(
+    // registerSigner registers new signer.
+    function registerSigner(address _signer)
+        external
+        onlyOwner
+    {
+        signers[_signer] = true;
+    }
+
+    // deregisterSigner deregisters the signer.
+    function deregisterSigner(address _signer)
+        external
+        onlyOwner
+    {
+        delete signers[_signer];
+    }
+
+    // setSignerThreshold sets signer threshold.
+    function setSignerThreshold(TransactionType _txType, uint64 _threshold)
+        external
+        onlyOwner
+    {
+        signerThresholds[uint64(_txType)] = _threshold;
+    }
+
+    function updateNonce(uint64 _requestNonce) internal {
+        if (_requestNonce > maxHandledNonce) {
+            maxHandledNonce = _requestNonce;
+        }
+        // TODO-Klaytn-ServiceChain: optimize this loop if possible.
+        for (uint64 i = sequentialHandledNonce; i <= maxHandledNonce; i++) {
```

Comment:
Since `_requestNonce` is a single value, it will not increase `sequentialHandledNonce` more than 1. 
Therefore, single comparision is enough, i think.
Also, it will be better if `handledNonces` is also updated in this function.
```solidity
handledNonces[requestNonce] = true;
if (requestNonce == sequentialHandledNonce+1) {
    sequentialHandledNonce++;
}
```

Reply:
Multiple transactions with random-ordered _requestNonce will be executed by the operator nodes. Therefore we need to check some range to check the sequential handle nonce.

Reply:
> Also, it will be better if handledNonces is also updated in this function.
Fixed. PTAL.

Reply:
I see. Thank you for the explanation and the change!